packages/node_modules/@ciscospark/redux-module-activity/src/actions.js

```diff
@@ -218,8 +218,10 @@ export function submitActivity(conversation, activity, user, spark) {
  * @returns {function}
  */
 export function setUserTyping(isTyping, conversation, spark) {
-  return (dispatch) =>
-    spark.internal.conversation.updateTypingStatus(conversation.toJS(), {typing: isTyping})
+  return (dispatch, getState) => {
+    const activity = getState().activity;
+    if (activity.getIn([`status`, `isTyping`]) !== isTyping) {
+      return spark.internal.conversation.updateTypingStatus(conversation.toJS(), {typing: isTyping})
```

Comment:
We have to keep sending this because the server will auto-timeout and send a stop typing event if we don't.

Reply:
Do we know what the timeout is?

Reply:
I can't find it documented anywhere.

Reply:
Ok, just got confirmation from the service owners that there is no auto-timeout. Clients handle all of this on their own.